pymeasure/adapters/visa.py

```diff
@@ -126,43 +128,66 @@ def __init__(self, resource_name, visa_library='', preprocess_reply=None,
+            This closes the connection to the resource for all adapters using
+            it currently (e.g. different adapters using the same GPIB line).
+        """
+        super().close()
+        try:
+            self.manager.close()
+        except AttributeError:
+            pass  # Closed from another adapter using the same connection.
+
     def _write(self, command, **kwargs):
         """Write a string command to the instrument appending `write_termination`.
 
         :param str command: Command string to be sent to the instrument
             (without termination).
-        :param kwargs: Keyword arguments for the connection itself.
+        :param \\**kwargs: Keyword arguments for the connection itself.
         """
         self.connection.write(command, **kwargs)
 
     def _write_bytes(self, content, **kwargs):
         """Write the bytes `content` to the instrument.
 
         :param bytes content: The bytes to write to the instrument.
-        :param kwargs: Keyword arguments for the connection itself.
+        :param \\**kwargs: Keyword arguments for the connection itself.
         """
         self.connection.write_raw(content, **kwargs)
 
     def _read(self, **kwargs):
         """Read up to (excluding) `read_termination` or the whole read buffer.
 
-        :param kwargs: Keyword arguments for the connection itself.
+        :param \\**kwargs: Keyword arguments for the connection itself.
         :returns str: ASCII response of the instrument (excluding read_termination).
         """
         return self.connection.read(**kwargs)
 
-    def _read_bytes(self, count, **kwargs):
+    def _read_bytes(self, count, break_on_termchar=False, **kwargs):
         """Read a certain number of bytes from the instrument.
 
         :param int count: Number of bytes to read. A value of -1 indicates to
-            read the whole read buffer.
-        :param kwargs: Keyword arguments for the connection itself.
+            read from the whole read buffer.
+        :param bool break_on_termchar: Stop reading at a termination character.
+        :param \\**kwargs: Keyword arguments for the connection itself.
         :returns bytes: Bytes response of the instrument (including termination).
         """
-        if count == -1:
-            return self.connection.read_raw(**kwargs)
+        if count >= 0:
+            return self.connection.read_bytes(count, break_on_termchar=break_on_termchar, **kwargs)
+        elif break_on_termchar:
+            return self.connection.read_raw(None, **kwargs)
         else:
-            return self.connection.read_bytes(count, **kwargs)
+            read_termination = self.connection.read_termination
+            self.connection.read_termination = None
+            # Try except allows to set the read_termination even after an error.
+            try:
+                return self.connection.read_raw(**kwargs)
+            finally:
+                self.connection.read_termination = read_termination
```

Comment:
This could be solved with a "read_termination_context" more elegantly, but there is an upstream bug in pyvisa: https://github.com/pyvisa/pyvisa/issues/723

Reply:
Good idea! I'd say file an issue here with the context, mention the upstream issue, then it's easy to check when we can start using this.